components/net/http_loader.rs

```diff
@@ -242,56 +239,67 @@ fn is_schemelessy_same_site(site_a: &ImmutableOrigin, site_b: &ImmutableOrigin)
 }
 
 /// <https://w3c.github.io/webappsec-referrer-policy/#strip-url>
-fn strip_url(mut referrer_url: ServoUrl, origin_only: bool) -> Option<ServoUrl> {
+fn strip_url_for_use_as_referrer(mut url: ServoUrl, origin_only: bool) -> Option<ServoUrl> {
     const MAX_REFERRER_URL_LENGTH: usize = 4096;
```

Comment:
I'm wondering why `url` always seems to be non-null, where are we dealing with Step 1?

Reply:
hmm, it doesn't seem clear to me [from the spec](https://w3c.github.io/webappsec-referrer-policy/#determine-requests-referrer) when a `referrerSource` can be null

Reply:
Ok, I have filed https://github.com/w3c/webappsec-referrer-policy/issues/139

Reply:
Thanks, looks like step 1 here is redundant.